lib/ex_aws/elasticache.ex

```diff
@@ -0,0 +1,595 @@
+defmodule ExAws.ElastiCache do
+  @moduledoc """
+  Operations for ElastiCache
+
+  Documentation: http://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/Welcome.html
+
+  ## Basic Usage
+  ```elixir
+  alias ExAws.ElastiCache
+
+  ElastiCache.create_cache_cluster("MyMemcachedCluster", "cache.t3.medium", "memcached", 1,
+    [preferred_availability_zones: ["us-east-1a"]]
+  ) |> ExAws.request
+
+  ElastiCache.describe_cache_clusters(
+    [cache_cluster_id: "MyMemcachedCluster", show_cache_node_info: true]
+  ) |> ExAws.request
+  ```
+  """
+
+  use ExAws.Utils,
+  format_type: :xml,
+  non_standard_keys: %{
+    az_mode: "AZMode"
+  }
+
+  @version "2015-02-02"
+
+  @type tag :: {key :: atom, value :: binary}
+  @type node_group_configuration ::
+    {
+      primary_availability_zone :: binary,
+      replica_availability_zones :: [binary, ...],
+      replica_count :: integer,
+      slots :: binary
+    }
```

Comment:
I feel like a map might be a better choice here. It's slightly more verbose but it doesn't require folks remembering what order each of these things are in.